cosi.proto

```diff
@@ -0,0 +1,131 @@
+syntax = "proto3";
+package cosi.v1alpha1;
+
+import "google/protobuf/descriptor.proto";
+
+option go_package = "cosi";
+
+extend google.protobuf.EnumOptions {
+    // Indicates that this enum is OPTIONAL and part of an experimental
+    // API that may be deprecated and eventually removed between minor
+    // releases.
+    bool alpha_enum = 1060;
+}
+
+extend google.protobuf.EnumValueOptions {
+    // Indicates that this enum value is OPTIONAL and part of an
+    // experimental API that may be deprecated and eventually removed
+    // between minor releases.
+    bool alpha_enum_value = 1060;
+}
+
+extend google.protobuf.FieldOptions {
+    // Indicates that a field MAY contain information that is sensitive
+    // and MUST be treated as such (e.g. not logged).
+    bool cosi_secret = 1059;
+
+    // Indicates that this field is OPTIONAL and part of an experimental
+    // API that may be deprecated and eventually removed between minor
+    // releases.
+    bool alpha_field = 1060;
+}
+
+extend google.protobuf.MessageOptions {
+    // Indicates that this message is OPTIONAL and part of an experimental
+    // API that may be deprecated and eventually removed between minor
+    // releases.
+    bool alpha_message = 1060;
+}
+
+extend google.protobuf.MethodOptions {
+    // Indicates that this method is OPTIONAL and part of an experimental
+    // API that may be deprecated and eventually removed between minor
+    // releases.
+    bool alpha_method = 1060;
+}
+
+extend google.protobuf.ServiceOptions {
+    // Indicates that this service is OPTIONAL and part of an experimental
+    // API that may be deprecated and eventually removed between minor
+    // releases.
+    bool alpha_service = 1060;
+}
+
+service Provisioner {
+    rpc ProvisionerGetInfo (ProvisionerGetInfoRequest) returns (ProvisionerGetInfoResponse) {}
+
+    rpc ProvisionerCreateBucket (ProvisionerCreateBucketRequest) returns (ProvisionerCreateBucketResponse) {}
+    rpc ProvisionerDeleteBucket (ProvisionerDeleteBucketRequest) returns (ProvisionerDeleteBucketResponse) {}
+
+    rpc ProvisionerGrantBucketAccess (ProvisionerGrantBucketAccessRequest) returns (ProvisionerGrantBucketAccessResponse);
+    rpc ProvisionerRevokeBucketAccess (ProvisionerRevokeBucketAccessRequest) returns (ProvisionerRevokeBucketAccessResponse);
+}
+
```

Comment:
This is also a general comment.  Since you don't have a file like spec.md in csi spec, please add general comments about each RPC here.  You have those in the KEP, but it is good to add here too so users of COSI can go thru cosi.proto and understand how to use the PRCs.

Reply:
I have started writing a spec.md but that will take sometime. Updating the details here for now.

Reply:
Sounds good.  In CSI spec repo, we only need to update spec.md and csi.proto will be generated as well.